src/test/ipv-stub/ipv-authorize.test.ts

```diff
@@ -0,0 +1,46 @@
+
+describe("IPV Authorize", () => {
+  const api = supertest("http://127.0.0.1:3000/");
+
+  it("should return 200 for valid GET request", async () => {
+    const response = await api
+      .get("/authorize")
+      .set("Authorization", "Bearer " + ENCRYPTED_JWT);
+    expect(response.statusCode).toBe(200);
+  });
+
+  it("should return 302 for valid POST request", async () => {
+    const response = await api
+      .post("/authorize")
+      .set("Authorization", "Bearer " + ENCRYPTED_JWT);
+    expect(response.statusCode).toBe(302);
+    expect(response.headers.location).toBe(
+      "https://oidc.sandpit.account.gov.uk/ipv-callback?code=12345"
+    );
+  });
+
+  it("should return 400 if token is not present in authorize request", async () => {
+    const response = await api.get("/authorize");
+    expect(response.statusCode).toBe(400);
+  });
+
+  it("should return 500 for invalid token in request", async () => {
+    const invalidJwt = "invalid.jwt.token";
+    const response = await api
+      .get("/authorize")
+      .set("Authorization", "Bearer " + invalidJwt);
+    expect(response.statusCode).toBe(500);
+  });
+
+  it("should return 500 if decryption fails", async () => {
```

Comment:
Is there a reason this is identical to the one above?

Reply:
good spot thanks